### PR TITLE
fix issue with starting requested block number was not included itself

### PR DESCRIPTION
**Pull request by NikVolf**
https://github.com/ethcore/parity/issues/511

### CODE REVIEW

File: sync/src/chain.rs [hunk 1]
```diff
@@ -927,13 +927,13 @@ impl ChainSync {
...
 		} else {
-			number = max(1, number);
+			number = max(0, number);
 		}
 		let max_count = min(MAX_HEADERS_TO_SEND, max_headers);
 		let mut count = 0;
 		let mut data = Bytes::new();
 		let inc = (skip + 1) as BlockNumber;
-		while number <= last && number > 0 && count < max_count {
+		while number <= last && number >= 0 && count < max_count {
```
Comment (tomusdrw) [inline sync/src/chain.rs]: `if number <= inc` couple of lines below should be `if number < inc` if when reverse blocks are requested they should also include 0.

Comment (NikVolf) [inline sync/src/chain.rs]: yeah writing a test thnx

**Comment by NikVolf** [comment]
@tomusdrw looks like test speaks that should be no such change
i think reverse request was buggy only for sequence ending with genesis

**Comment by arkpar** [comment]
Forward request was "buggy" only for genesis block as well. The original idea was that we should not return genesis block anyway. But if the spec demands it, so be it

**Comment by NikVolf** [comment]
@arkpar right

**Pull request merged**